systemtest/src/main/java/io/strimzi/systemtest/bases/IMetrics.java

```diff
@@ -0,0 +1,121 @@
+import static io.strimzi.test.BaseITST.kubeClient;
+
+public interface IMetrics {
+
+    Logger LOGGER = LogManager.getLogger(IMetrics.class);
+    Object LOCK = new Object();
+
+    /**
+     * Collect metrics from specific pod
+     * @param podName pod name
+     * @return collected metrics
+     */
+    default String collectMetrics(String podName, String metricsPath) throws InterruptedException, ExecutionException, IOException {
+        ArrayList<String> command = new ArrayList<>();
+        command.add("curl");
+        command.add(kubeClient().getPod(podName).getStatus().getPodIP() + ":9404" + metricsPath);
+        ArrayList<String> executableCommand = new ArrayList<>();
+        executableCommand.addAll(Arrays.asList(cmdKubeClient().toString(), "exec", podName, "-n", kubeClient().getNamespace(), "--"));
+        executableCommand.addAll(command);
+
+        Exec exec = new Exec();
+        // 20 seconds should be enough for collect data from the pod
+        int ret = exec.execute(null, executableCommand, 20_000);
+
+        synchronized (LOCK) {
+            LOGGER.info("Metrics collection for pod {} return code - {}", podName, ret);
+        }
+
+        return exec.out();
+    }
+
+    default HashMap<String, String> collectKafkaPodsMetrics(String clusterName) {
+        LabelSelector kafkaSelector = kubeClient().getStatefulSetSelectors(KafkaResources.kafkaStatefulSetName(clusterName));
+        return collectMetricsFromPods(kafkaSelector);
+    }
+
+    default HashMap<String, String> collectZookeeperPodsMetrics(String clusterName) {
+        LabelSelector zookeeperSelector = kubeClient().getStatefulSetSelectors(KafkaResources.zookeeperStatefulSetName(clusterName));
+        return collectMetricsFromPods(zookeeperSelector);
+    }
+
+    default HashMap<String, String> collectKafkaConnectPodsMetrics(String clusterName) {
+        LabelSelector connectSelector = kubeClient().getDeploymentSelectors(KafkaConnectResources.deploymentName(clusterName));
+        return collectMetricsFromPods(connectSelector);
+    }
+
+    default HashMap<String, String> collectKafkaExporterPodsMetrics(String clusterName) {
+        LabelSelector connectSelector = kubeClient().getDeploymentSelectors(KafkaExporterResources.deploymentName(clusterName));
+        return collectMetricsFromPods(connectSelector, "/metrics");
+    }
+
+
+    /**
+     * Parse out specific metric from whole metrics file
+     * @param pattern regex patern for specific metric
+     * @param data all metrics data
+     * @return list of parsed values
+     */
+    default ArrayList<Double> collectSpecificMetric(Pattern pattern, HashMap<String, String> data) {
+        ArrayList<Double> values = new ArrayList<>();
+
+        data.forEach((k, v) -> {
+            Matcher t = pattern.matcher(v);
+            if (t.find()) {
+                values.add(Double.parseDouble(t.group(1)));
+            }
+        });
+        return values;
```

Comment:
There's little point in using `forEach` like this; better to use a `for` statement which doesn't require a lambda. Lambdas come with a cost so just using `forEach` all over the place where a `for` stmt would do just as well isn't a good habit to get into.

It's a balance of course, and often I find the simplicity of the Stream API wins, but it's worth looking especially critically at uses for `forEach`, because they're often trivially written as `for`.